packages/opentelemetry-exporter-zipkin/src/types.ts

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * Exporter config
+ */
+export interface ExporterConfig {
+  logger?: types.Logger;
+  serviceName: string;
+  url?: string;
+  // Initiates a request with spans in memory to the backend.
+  forceFlush?: boolean;
+  // Optional mapping overrides for OpenTelemetry status code and description.
+  statusCodeTagName?: string;
+  statusDescriptionTagName?: string;
+}
+
+/**
+ * Zipkin Span
+ * @see https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml
+ */
+export interface Span {
+  /**
+   * Trace identifier, set on all spans within it.
+   */
+  traceId: string;
+  /**
+   * The logical operation this span represents in lowercase (e.g. rpc method).
+   * Leave absent if unknown.
+   */
+  name: string;
+  /**
+   * The parent span ID or absent if this the root span in a trace.
+   */
+  parentId?: string;
+  /**
+   * Unique 64bit identifier for this operation within the trace.
+   */
+  id: string;
+  /**
+   * When present, kind clarifies timestamp, duration and remoteEndpoint.
+   * When absent, the span is local or incomplete.
+   */
+  kind?: SpanKind;
+  /**
+   * Epoch microseconds of the start of this span, possibly absent if
+   * incomplete.
+   */
+  timestamp: number;
+  /**
+   * Duration in microseconds of the critical path, if known.
+   */
+  duration: number;
+  /**
+   * True is a request to store this span even if it overrides sampling policy.
+   * This is true when the `X-B3-Flags` header has a value of 1.
+   */
+  debug?: boolean;
+  /**
+   * True if we are contributing to a span started by another tracer (ex on a
+   * different host).
+   */
+  shared?: boolean;
+  /**
+   * The host that recorded this span, primarily for query by service name.
+   */
+  localEndpoint: Endpoint;
+  /**
+   * Associates events that explain latency with the time they happened.
+   */
+  annotations?: Annotation[];
+  /**
+   * Tags give your span context for search, viewing and analysis.
+   */
+  tags: Tags;
+  /**
+   * TODO: `remoteEndpoint`, do we need to support it?
+   * When an RPC (or messaging) span, indicates the other side of the
+   * connection.
+   */
+}
+
+/**
+ * Associates an event that explains latency with a timestamp.
+ * Unlike log statements, annotations are often codes. Ex. "ws" for WireSend
+ * Zipkin v1 core annotations such as "cs" and "sr" have been replaced with
+ * Span.Kind, which interprets timestamp and duration.
+ */
+export interface Annotation {
+  /**
+   * Epoch microseconds of this event.
+   * For example, 1502787600000000 corresponds to 2017-08-15 09:00 UTC
+   */
+  timestamp: number;
+  /**
+   * Usually a short tag indicating an event, like "error"
+   * While possible to add larger data, such as garbage collection details, low
+   * cardinality event names both keep the size of spans down and also are easy
+   * to search against.
+   */
+  value: string;
+}
+
+/**
+ * The network context of a node in the service graph.
+ */
+export interface Endpoint {
```

Comment:
Zipkin forma also supports ipv6 for endpoints. Any particular reason for not including it?

Reply:
Yes, you're right. This is what I found on zipkin API.

description: | The network context of a node in the service graph
-- | --
serviceName | stringLower-case label of this node in the service graph, such as "favstar". Leaveabsent if unknown.This is a primary label for trace lookup and aggregation, so it should beintuitive and consistent. Many use a name from service discovery.
ipv4 | string($ipv4)The text representation of the primary IPv4 address associated with thisconnection. Ex. 192.168.99.100 Absent if unknown.
ipv6 | string($ipv6)The text representation of the primary IPv6 address associated with aconnection. Ex. 2001:db8::c001 Absent if unknown.Prefer using the ipv4 field for mapped addresses.
port | integerDepending on context, this could be a listen port or the client-side of asocket. Absent if unknown. Please don’t set to zero.

@hekike Could you please add support `ipv6`? Thanks

Reply:
I'd expect ipv6 would be handled by node, do I miss something?
```
$ node -p -e 'require("url").parse("http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:9411/api/v2/spans")'
Url {
  protocol: 'http:',
  slashes: true,
  auth: null,
  host: '[fedc:ba98:7654:3210:fedc:ba98:7654:3210]:9411',
  port: '9411',
  hostname: 'fedc:ba98:7654:3210:fedc:ba98:7654:3210',
  hash: null,
  search: null,
  query: null,
  pathname: '/api/v2/spans',
  path: '/api/v2/spans',
  href:
   'http://[fedc:ba98:7654:3210:fedc:ba98:7654:3210]:9411/api/v2/spans' }
```